controllers/storagecluster/reconcile.go

```diff
@@ -152,6 +154,11 @@ func (r *StorageClusterReconciler) Reconcile(ctx context.Context, request reconc
 			// Request object not found, could have been deleted after reconcile request.
 			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
 			// Return and don't requeue
+			operatorConditionErr := r.OperatorCondition.Set(ctx, "True", conditions.Option(conditions.WithMessage("StorageCluster is deleted")), conditions.Option(conditions.WithReason("notFound")))
+			if operatorConditionErr != nil {
+				r.Log.Error(operatorConditionErr, "Unable to update OperatorCondition")
+				return reconcile.Result{}, operatorConditionErr
+			}
```

Comment:
Given that this seems to be a repeated pattern, it might make sense to put it in its own function. In particular, issuing a `return` here prevents the logging of any previous error you're trying to not overwrite. So passing in a per-existing error (or `nil` in this case) should allow you to properly log both of them. Mostly I want to see if you can avoid adding so many `return` statements.

Reply:
Made the changes. Let me know if you have some other suggestions